projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/VdkTextInput.tsx

```diff
@@ -1,75 +1,121 @@
+     */
+    label: string;
 
-/**
-   * This function is used to be sure that placeholder text is not overflowing in case of long path
-   * if the string that is given is not path it will return the same string
-   * if it is a path it will return the last directory
-   */
-const  getLastPartOfPath = (path: string): string => {
-  const pathParts = path.split(/(?=[/\\])/); // Lookahead assertion to keep delimiter
-  return pathParts[pathParts.length - 1];
+    /**
+     * Callback for when the width of the input is computed
+     */
+    onWidthComputed?: (width: number) => void;
 }
 
 export default class VDKTextInput extends Component<IVdkTextInputProps> {
-  /**
-   * Returns a React component for rendering a div with input and value  for VDK Option.
-   *
-   * @param props - component properties
-   * @returns React component
-   */
-  constructor(props: IVdkTextInputProps) {
-    super(props);
-  }
-  /**
-   * Renders a div with input and value  for VDK Option.
-   *
-   * @returns React element
-   */
-  render(): React.ReactElement {
-    return (
-      <>
-        <div className="jp-vdk-input-wrapper">
-          <label className="jp-vdk-label" htmlFor={this.props.option}>
-            {this.props.label}
-          </label>
-          <input
-            type="text"
-            id={this.props.option}
-            className="jp-vdk-input"
-            placeholder={getLastPartOfPath(this.props.value)}
-            onChange={this.onInputChange}
-          />
-        </div>
-      </>
-    );
-  }
-  /**
-   * Callback invoked upon changing the input.
-   *
-   * @param event - event object
-   */
-  private onInputChange = (event: any): void => {
-    const nameInput = event.currentTarget as HTMLInputElement;
-    let value = nameInput.value;
-    if (!value) value = this.props.value;
-    jobData.set(this.props.option, value);
-  };
+    private inputRef: RefObject<HTMLInputElement> = React.createRef();
+
+    /**
+     * Initializes a new instance of the VDKTextInput component.
+     *
+     * @param props - component properties
+     */
+    constructor(props: IVdkTextInputProps) {
+        super(props);
+    }
+
+    /**
+     * Lifecycle method called after the component has mounted.
+     */
+    componentDidMount() {
+        this.adjustInputWidth();
+    }
+
+    /**
+     * Lifecycle method called after the component has updated.
+     *
+     * @param prevProps - previous component properties
+     */
+    componentDidUpdate(prevProps: IVdkTextInputProps): void {
+        if (prevProps.value !== this.props.value) {
+            this.adjustInputWidth();
+        }
+    }
+
+    /**
+     * Adjusts the input width based on its placeholder value.
+     */
+     adjustInputWidth(): void {
+         const currentInput = this.inputRef.current;
+         if (!currentInput) return;
+
+         let maxWidth = 250;
+
+         jobData.forEach((value) => {
+             const tempSpan = document.createElement('span');
```

Comment:
@antoniivanov the idea of the method is to change the input size according to the placeholder we have, you can check the photo i have provided in the description. 
The inputs are not responsive by default and if we have a really long path as in the example it would hide the placeholder which looks strange
My idea here was the user to be able to see the whole path so they know what kind of path we expect from them since in the usability tests it was a great issue 
Do you have any other recommendations for this issue? Actually yes manipulating the dom is not bad but in the worst case scenario we would get the default input size

Reply:
@antoniivanov I have modified the code to use React's state for adjusting the input width. Now we use the DOM only for the tempSpan and if something happens it will use the default 250px value